examples/nginx-grafana-metabase/README.md

```diff
@@ -0,0 +1,162 @@
+
+### Nginx Access Log Format
+Nginx's [access log format](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/#setting-up-the-access-log) is specified in `/etc/nginx/nginx.conf` and it broadly looks like this:
+```bash
+http {
+    log_format compression '$remote_addr - $remote_user [$time_local] '
+                           '"$request" $status $body_bytes_sent '
+                           '"$http_referer" "$http_user_agent"';
+
+    server {
+        access_log /var/log/nginx/access.log;
+        ...
+    }
+}
+```
+
+Below is a brief explanation of how each access log line is parsed into 12 columns:
+* `$remote_addr` is stored in column `remote_ip` of type `ipv4`;
+* `-` is stored in column `rfc1413_ident` of type `string`;
+* `$remote_user` is stored in column `remote_user` of type `string`;
+* `$time_local` os stored in column `date_time` of type `datetime64`;
+* `$request` is split into 4 columns each of type `string` to make our analysis easier:
+  * column `http_verb` of type `string`;
+  * column `path` of type `string` and;
+  * column `http_ver` of type `string`;
+  * column `malicious_request` of type `string` for any `$request` that fails to parse correctly; 
+* `$status` is stored in column `status` of type `int`;
+* `$body_bytes_sent` is stored in column `size` of type `int`;
+* `$http_referer` is stored in column `referer` of type `string`;
+* `$http_user_agent` is stored in column `user_agent` of type `string`.
+
+The previous article created a stream with a total of 12 columns to take advantage of Timeplus Proton's fast aggregation for our analysis. We will reuse the same approach in this article. 
+
+With that out of the way, let's dive into this article proper!
+
+
+## Docker Environment
+The [`docker-compose.yaml`](docker-compose.yaml) file defines 3 containers:
+* a Timeplus Proton container;
+* a Grafana container;
+* a third container that seeds the Timeplus Proton container with access log data.
+
+### DDL Overview
+The 3rd container uses the `proton client` to seed the Timeplus Proton database with access log data from an SQL file: 
+* [`01_nginx_access_log.sql`](01_nginx_access_log.sql). 
+
+The SQL file contains the following DDL:
+![SQL code screenshot with line numbers](01_nginx_access_log.png)
+
+### Generating Data to Mimic Nginx Access Data
+On line 3, the [`CREATE RANDOM STREAM`](https://docs.timeplus.com/proton-create-stream#create-random-stream) DDL is used to create a random stream named `nginx_access_log`. 
+
+Just like in the previous article, the stream also has 12 columns. Each column is randomly assigned a `default` value as you can see on lines 4 - 30.
+
+Using the line numbers 4 - 30 as reference, the table below goes over each column and the database [functions](https://docs.timeplus.com/functions) used to generate test data.
+| Line  | Column | `default` | 
```

Comment:
yes, the table is nice. I guess there might be a better way to organize the content. Maybe have this table right after you explain the log format. Also the user won't really compare the image with the table, so having the line number column is not very helpful.